tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py

```diff
@@ -1985,6 +1960,185 @@ def visit_importfrom(self, node):
+        "C4752": (
+            "delete* or begin_delete* does not return None or LROPoller[None].",
+            "delete-operation-wrong-return-type",
+            "delete* or begin_delete* functions should return None or LROPoller[None]."
+        ),
+    }
+
+    def visit_functiondef(self,node):
+        """Visits all delete functions and checks that their return types
+        are LROPoller or None. """
+        try:
+            if node.name.startswith("delete") or node.name.startswith("begin_delete"):
+                inferred = node.infer_call_result()
+                for x in inferred:
+                    if str(x)=="Uninferable" or x == None or x.value == None or \
+                        x.name == "LROPoller" or x.name == "AsyncLROPoller":
+                            return
+                self.add_message(
+                    msgid=f"delete-operation-wrong-return-type",
+                    node=node,
+                    confidence=None,
+                )   
+        except:
+            pass
+    
+    visit_asyncfunctiondef = visit_functiondef
+
+
+class AsyncMethodsReturnAsyncIterables(BaseChecker):
+    __implements__ = IAstroidChecker
+
+    """Rule to check that an async method returns an async iterable."""
+    name = "async-return-async-iterable"
+    priority = -1
+    msgs = {
+        "C4753": (
+            "async method returns sync iterable.",
+            "async-return-async-iterable",
+            "async methods should return async iterables."
+        ),
+    }
+
+    def visit_asyncfunctiondef(self, node):
+        """Checks all async functions, if there is a return statement, checks that it is preceded by an await. 
+        If it is a call to a function, it checks that the function is async. i.e AsyncLROPoller"""
+        try:
+            if "Client" in node.parent.name:
+                for n in node.body:
+                    if isinstance(n, astroid.Return):
+                        if isinstance(n.value, astroid.Await):
+                            return 
+                        if isinstance(n.value, astroid.Call) and "async" not in n.value.func.as_string():
+                            self.add_message(
+                                msgid=f"async-return-async-iterable",
+                                node=node,
+                                confidence=None,
+                            )   
+                            
+        except:
+            pass
+
+
+class NoAzureCoreTracebackUseRaiseFrom(BaseChecker):
+    __implements__ = IAstroidChecker
+
+    """Rule to check that we don't use raise_with_traceback in azure core."""
+    name = "no-raise-with-traceback"
+    priority = -1
+    msgs = {
+        "C4754": (
+            "Don't use raise_with_traceback, use python 3 raise from syntax.",
+            "no-raise-with-traceback",
+            "Don't use raise_with_traceback instead use python 3 raise from syntax."
+        ),
+    }
+    AZURE_CORE= "azure.core.exceptions"
```

Comment:
looks unused... should this be used to check if it's being imported from that module?